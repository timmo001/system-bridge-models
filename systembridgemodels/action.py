"""Action."""
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass
class Action:
    """Action."""

    command: str
    data: dict[str, Any] | None = None
    label: str | None = None
