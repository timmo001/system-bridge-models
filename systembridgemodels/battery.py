# generated by datamodel-codegen:
#   filename:  battery.json

from __future__ import annotations

from pydantic import BaseModel, Extra, Field


class LastUpdated(BaseModel):
    """
    Last updated
    """

    is_charging: float | None = None
    percentage: float | None = None


class Battery(BaseModel):
    """
    Battery
    """

    class Config:
        extra = Extra.allow

    id: str | None = Field(None, description="Event ID")
    is_charging: bool | None = None
    percentage: float | None = None
    last_updated: LastUpdated | None = Field(None, description="Last updated")
