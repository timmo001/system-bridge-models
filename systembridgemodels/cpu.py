# generated by datamodel-codegen:
#   filename:  cpu.json

from __future__ import annotations

from pydantic import BaseModel, Extra, Field


class LastUpdated(BaseModel):
    """
    Last updated
    """

    count: float | None = None
    frequency_current: float | None = None
    frequency_min: float | None = None
    frequency_max: float | None = None
    load_average: float | None = None
    power_package: float | None = None
    stats_ctx_switches: float | None = None
    stats_interrupts: float | None = None
    stats_soft_interrupts: float | None = None
    stats_syscalls: float | None = None
    temperature: float | None = None
    times_user: float | None = None
    times_system: float | None = None
    times_idle: float | None = None
    times_interrupt: float | None = None
    times_dpc: float | None = None
    times_percent_user: float | None = None
    times_percent_system: float | None = None
    times_percent_idle: float | None = None
    times_percent_interrupt: float | None = None
    times_percent_dpc: float | None = None
    usage: float | None = None
    voltage: float | None = None


class Cpu(BaseModel):
    """
    CPU
    """

    class Config:
        extra = Extra.allow

    id: str | None = Field(None, description="Event ID")
    count: int | None = None
    frequency_current: float | None = None
    frequency_min: float | None = None
    frequency_max: float | None = None
    load_average: float | None = None
    power_package: float | None = None
    stats_ctx_switches: int | None = None
    stats_interrupts: int | None = None
    stats_soft_interrupts: int | None = None
    stats_syscalls: int | None = None
    temperature: float | None = None
    times_user: float | None = None
    times_system: float | None = None
    times_idle: float | None = None
    times_interrupt: float | None = None
    times_dpc: float | None = None
    times_percent_user: float | None = None
    times_percent_system: float | None = None
    times_percent_idle: float | None = None
    times_percent_interrupt: float | None = None
    times_percent_dpc: float | None = None
    usage: float | None = None
    voltage: float | None = None
    last_updated: LastUpdated | None = Field(None, description="Last updated")
