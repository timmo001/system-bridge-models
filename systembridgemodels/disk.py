# generated by datamodel-codegen:
#   filename:  disk.json

from __future__ import annotations

from pydantic import BaseModel, Extra, Field


class LastUpdated(BaseModel):
    """
    Last updated
    """

    devices: float
    io_counters_read_count: float | None = None
    io_counters_write_count: float | None = None
    io_counters_read_bytes: float | None = None
    io_counters_write_bytes: float | None = None
    io_counters_read_time: float | None = None
    io_counters_write_time: float | None = None
    partitions: float


class Disk(BaseModel):
    """
    Disk
    """

    class Config:
        extra = Extra.allow

    id: str | None = Field(None, description="Event ID")
    devices: list
    io_counters_read_count: int | None = None
    io_counters_write_count: int | None = None
    io_counters_read_bytes: int | None = None
    io_counters_write_bytes: int | None = None
    io_counters_read_time: int | None = None
    io_counters_write_time: int | None = None
    partitions: list
    last_updated: LastUpdated | None = Field(None, description="Last updated")
