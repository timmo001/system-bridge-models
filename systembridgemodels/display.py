# generated by datamodel-codegen:
#   filename:  display.json

from __future__ import annotations

from pydantic import BaseModel, Extra, Field


class LastUpdated(BaseModel):
    """
    Last updated
    """

    displays: float | None = None


class Display(BaseModel):
    """
    Display
    """

    class Config:
        extra = Extra.allow

    id: str | None = Field(None, description="Event ID")
    displays: list | None = None
    last_updated: LastUpdated | None = Field(None, description="Last updated")
