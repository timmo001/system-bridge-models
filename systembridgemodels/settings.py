"""System."""
from __future__ import annotations

from dataclasses import dataclass, field
from enum import Enum
from uuid import uuid4


def generate_token() -> str:
    """Generate token."""
    return str(uuid4())


class SettingAPIEnum(Enum):
    """Setting API Enum."""

    TOKEN = "token"
    PORT = "port"


class SettingHotkeyEnum(Enum):
    """Setting Hotkey Enum."""

    NAME = "name"
    KEY = "key"


class SettingDirectoryEnum(Enum):
    """Setting Media Directory Enum."""

    NAME = "name"
    PATH = "path"


class SettingMediaEnum(Enum):
    """Setting Media Enum."""

    DIRECTORIES = "directories"


class SettingEnum(Enum):
    """Setting Enum."""

    API = "api"
    AUTOSTART = "autostart"
    KEYBOARD_HOTKEYS = "keyboard_hotkeys"
    LOG_LEVEL = "log_level"
    MEDIA = "media"


@dataclass
class SettingsAPI:
    """Settings API."""

    token: str = field(default_factory=generate_token)
    port: int = field(default=9174)


@dataclass
class SettingHotkey:
    """Setting Hotkey."""

    name: str
    key: str


@dataclass
class SettingDirectory:
    """Setting Directory."""

    name: str
    path: str


@dataclass
class SettingsMedia:
    """Settings Media."""

    directories: list[SettingDirectory] = field(default_factory=lambda: [])


@dataclass
class Settings:
    """Settings."""

    api: SettingsAPI = field(default_factory=SettingsAPI)
    autostart: bool = field(default=False)
    keyboard_hotkeys: list[SettingHotkey] = field(default_factory=lambda: [])
    log_level: str = field(default="INFO")
    media: SettingsMedia = field(default_factory=SettingsMedia)
