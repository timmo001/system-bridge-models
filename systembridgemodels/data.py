"""Data"""
from __future__ import annotations

from dataclasses import dataclass
from typing import Any

from .battery import Battery
from .cpu import CPU

# from .disk import Disk
# from .display import Display
# from .media import Media
# from .memory import Memory
# from .network import Network
# from .processes import Processes
# from .system import System


@dataclass
class Data:
    """Data"""

    battery: Battery | None = None
    cpu: CPU | None = None
    # disk: Disk | None = None
    # display: Display | None = None
    # gpu: dict[str, Any] | None = None
    # media: Media | None = None
    # memory: Memory | None = None
    # network: Network | None = None
    # processes: Processes | None = None
    # sensors: dict[str, Any] | None = None
    # system: System | None = None
