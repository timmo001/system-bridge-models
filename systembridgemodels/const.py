# Model
from .battery import Battery
from .cpu import Cpu
from .data import Data
from .disk import Disk
from .display import Display
from .generic import Generic
from .gpu import Gpu
from .keyboard_key import KeyboardKey
from .keyboard_text import KeyboardText
from .media import Media
from .media_directories import MediaDirectories
from .media_files import File, MediaFiles
from .memory import Memory
from .network import Network
from .notification import Notification
from .open_path import OpenPath
from .open_url import OpenUrl
from .processes import Processes
from .response import Response
from .sensors import Sensors
from .system import System

MODEL_BATTERY = "battery"
MODEL_CPU = "cpu"
MODEL_DATA = "data"
MODEL_DISK = "disk"
MODEL_DISPLAY = "display"
MODEL_GENERIC = "generic"
MODEL_GPU = "gpu"
MODEL_KEYBOARD_KEY = "keyboard_key"
MODEL_KEYBOARD_TEXT = "keyboard_text"
MODEL_MEDIA = "media"
MODEL_MEDIA_DIRECTORIES = "media_directories"
MODEL_MEDIA_FILE = "media_file"
MODEL_MEDIA_FILES = "media_files"
MODEL_MEMORY = "memory"
MODEL_NETWORK = "network"
MODEL_NOTIFICATION = "notification"
MODEL_OPEN_PATH = "open_path"
MODEL_OPEN_URL = "open_url"
MODEL_PROCESSES = "processes"
MODEL_RESPONSE = "response"
MODEL_SECRETS = "secrets"
MODEL_SENSORS = "sensors"
MODEL_SETTINGS = "settings"
MODEL_SYSTEM = "system"

MODEL_MAP = {
    MODEL_BATTERY: Battery,
    MODEL_CPU: Cpu,
    MODEL_DATA: Data,
    MODEL_DISK: Disk,
    MODEL_DISPLAY: Display,
    MODEL_GENERIC: Generic,
    MODEL_GPU: Gpu,
    MODEL_KEYBOARD_KEY: KeyboardKey,
    MODEL_KEYBOARD_TEXT: KeyboardText,
    MODEL_MEDIA_DIRECTORIES: MediaDirectories,
    MODEL_MEDIA_FILE: File,
    MODEL_MEDIA_FILES: MediaFiles,
    MODEL_MEDIA: Media,
    MODEL_MEMORY: Memory,
    MODEL_NETWORK: Network,
    MODEL_NOTIFICATION: Notification,
    MODEL_OPEN_PATH: OpenPath,
    MODEL_OPEN_URL: OpenUrl,
    MODEL_PROCESSES: Processes,
    MODEL_RESPONSE: Response,
    MODEL_SENSORS: Sensors,
    MODEL_SYSTEM: System,
}
