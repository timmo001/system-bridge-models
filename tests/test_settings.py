"""Test the settings model."""

from systembridgemodels.settings import Settings, SettingsAPI, SettingsMedia


def test_settings():
    """Test the settings."""
    settings = Settings()
    assert isinstance(settings, Settings)
    assert isinstance(settings.api, SettingsAPI)
    assert isinstance(settings.api.token, str)
    assert isinstance(settings.api.port, int)
    assert isinstance(settings.autostart, bool)
    assert isinstance(settings.keyboard_hotkeys, list)
    assert isinstance(settings.log_level, str)
    assert isinstance(settings.media, SettingsMedia)
    assert isinstance(settings.media.directories, list)
