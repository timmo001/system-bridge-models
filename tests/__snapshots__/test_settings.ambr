# serializer version: 1
# name: test_settings
  Settings(api=SettingsAPI(token='token', port=9170), autostart=False, keyboard_hotkeys=[], log_level='INFO', media=SettingsMedia(directories=[]))
# ---
